Model/src/main/java/org/eupathdb/common/service/CyberSourceFormService.java

```diff
@@ -0,0 +1,169 @@
+        .put("reference_number", referenceNumber)
+        .put("transaction_type", "authorization")
+        .put("transaction_uuid", UUID.randomUUID().toString())
+        .put("profile_id", profileId)
+        .put("access_key", accessKey)
+        .put("amount", amount)
+        .put("currency", currency)
+        .toMap();
+
+    // calculate signature from the exising fields and add to the form map
+    paramMap.put("signature", getSignature(paramMap, secretKey));
+
+    return Response.ok(new JSONObject(paramMap).toString()).build();
+  }
+
+  private static void logFormGeneration(User requestingUser, String referenceNumber, String amount, String currency) {
+    LOG.info(Arrays.stream(new String[] {
+        String.valueOf(requestingUser.getUserId()),
+        "guest=" + requestingUser.isGuest(),
+        referenceNumber,
+        amount,
+        currency
+    }).map(s -> "\t" + s).collect(Collectors.joining()));
+  }
+
+  private static String getSignature(Map<String,String> params, String secretKey) {
+
+    // aggregate data into comma-delimited key=value pairs
+    String data = Arrays
+        .stream(params.get("signed_field_names").split(","))
+        .map(name -> name + "=" + params.get(name))
+        .collect(Collectors.joining(","));
+
+    // generate the signature and return
+    try {
+      SecretKeySpec secretKeySpec = new SecretKeySpec(secretKey.getBytes(), HMAC_SHA256);
+      Mac mac = Mac.getInstance(HMAC_SHA256);
+      mac.init(secretKeySpec);
+      byte[] rawHmac = mac.doFinal(data.getBytes(StandardCharsets.UTF_8));
+      return new String(Base64.getEncoder().encode(rawHmac), StandardCharsets.UTF_8);
+    }
+    catch (InvalidKeyException | NoSuchAlgorithmException e) {
+      throw new WdkRuntimeException("Unable to sign cybersource form", e);
+    }
+  }
+
+  private static String getRandomDigits(int numDigits) {
+    Random rand = new Random();
+    return IntStream.range(0, numDigits)
+        .mapToObj(i -> String.valueOf(rand.nextInt(10)))
+        .collect(Collectors.joining());
+  }
+
+  static String validateAmountParam(String amount) {
+    if (amount == null || !MONEY_PATTERN.matcher(amount).matches()) {
+      throw new BadRequestException("'amount' parameter is required and represent a numeric payment amount in US Dollars.");
+    }
+    return amount.indexOf(".") == -1 ? amount + ".00" : amount;
+  }
+
+  // TODO: in the future we will probably support multiple currencies
+  static String validateCurrencyParam(String currency) {
+    if (currency != null && !currency.toUpperCase().equals("USD")) {
+      throw new BadRequestException("'currency' parameter, if passed, must be 'USD'; other currencies are not yet supported");
+    }
+    return "USD";
+  }
+
+  private static String getUTCDateTime() {
+    SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss'Z'");
+    sdf.setTimeZone(TimeZone.getTimeZone("UTC"));
+    return sdf.format(new Date());
```

Comment:
The more "modern" approach to get an RFC3339 timestamp would probably be one of:
```java
OffsetDateTime.now(ZoneOffset.UTC).format(DateTimeFormatter.ISO_INSTANT)
OffsetDateTime.now(ZoneOffset.UTC).truncatedTo(ChronoUnit.SECONDS).format(DateTimeFormatter.ISO_INSTANT)
```